src/pages/index.js

```diff
@@ -67,36 +95,151 @@ addNewCardButton.addEventListener("click", () => {
+      console.error("info not updated", error);
+    })
+    .finally(() => {
+      profilePopup.setButtonText();
+    });
+});
 
-  userInfo.setUserInfo({ title: name, description: job });
-  profilePopup.close();
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  jobSelector: ".profile__description",
+  avatarSelector: ".avatar__image",
 });
 
 profilePopup.setEventListeners();
 
 profileEditButton.addEventListener("click", () => {
-  
-    const currentUserInfo = userInfo.getUserInfo();
-    nameInput.value = currentUserInfo.title;
-    jobInput.value = currentUserInfo.description;
-    editFormValidator.resetValidation();
-  
+  profilePopup.open();
+  const currentUserInfo = userInfo.getUserInfo();
+  nameInput.value = currentUserInfo.title;
+  jobInput.value = currentUserInfo.description;
+  editFormValidator.resetValidation();
+
   profilePopup.open();
 });
 
+//============================
+//=== Edit Avatar Popup ======
+//============================
+const avatarPopup = new PopupWithForm("#edit-avatar-modal", (formData) => {
+  const avatarUrl = formData.avatar;
+
+  if (!avatarUrl) {
+    console.error("URL missing");
+    return;
+  }
+  avatarPopup.setButtonText(true);
+  api
+    .updateAvatar(avatarUrl)
+    .then((avatarData) => {
+      userInfo.setUserAvatar(avatarData);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarPopup.setButtonText();
+    });
+});
+
+avatarPopup.setEventListeners();
+
+avatarEditButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+//======================
+//=== image Popup ======
+//======================
 const imagePopup = new PopupWithImage({
   popupSelector: "#preview-image-modal",
 });
 imagePopup.setEventListeners();
 
-const userInfo = new UserInfo({
-  profileName: ".profile__title",
-  jobElement: ".profile__description",
+//=========================
+//=== delete card Popup ===
+//=========================
+const deleteCardPopup = new PopupWithDelete({
+  popupSelector: "#delete-card-modal",
 });
 
+deleteCardPopup.setEventListeners();
+
+function handleDeleteClick(card) {
+  deleteCardPopup.open();
+
+  deleteCardPopup.setSubmitAction(() => {
+    deleteCardPopup.setButtonText(true);
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((error) => {
+        console.error("Error deleting card: ", error);
+      })
+      .finally(() => {
+        deleteCardPopup.setButtonText();
+      });
+  });
+}
 
-
-
+//==================================
+//=== removeLike/addLike to card ===
+//==================================
+function handleLikeClick(card) {
+  // if card isLiked removeLike
+  if (card._isLiked) {
+    api
+      .removeLike(card._id)
+      .then(() => {
+        card._handleLikeIcon();
+      })
+      .catch((error) => {
+        console.log("Error removing like: ", error);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors